Make sure that reviveOffers is sent after last decline

The reviveOffer call might be processed BEFORE a declineOffer that should be reconsidered.

We do not guarantee the order of the calls in Marathon. Even if we did, the `MesosSchedulerDriver` does not guarantee that the `declineOffer` and `reviveOffer` messages arrive at the Master in the order of the related call execution.

Since we do not get any message acknowledgement, I see the following approaches (may be combined):

A) insist on a reasonably low `--decline_offer_duration` for "fast enough" error recovery
B) delay `reviveOffer` calls for a while (one second?) or repeat them once after a while.

Sample execution from our integration tests:

```
[17:44:01][Step 3/3] [2015-09-03 15:44:01,647] INFO marathon: [2015-09-03 15:44:01,647] INFO Upgrade group id:/ version:2015-09-03T15:44:01.647Z with force:true (mesosphere.marathon.state.GroupManager:138) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,649] INFO marathon: [2015-09-03 15:44:01,649] INFO Computed new deployment plan: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,649] INFO marathon: DeploymentPlan 2015-09-03T15:44:01.647Z NO STEPS (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,649] INFO marathon:  (mesosphere.marathon.state.GroupManager:161) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,649] INFO marathon: [2015-09-03 15:44:01,649] INFO Deploy plan with force=true: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,649] INFO marathon: DeploymentPlan 2015-09-03T15:44:01.647Z NO STEPS (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,650] INFO marathon:   (mesosphere.marathon.MarathonSchedulerService$$EnhancerByGuice$$7a577b4d:106) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,663] INFO marathon: [2015-09-03 15:44:01,662] INFO Deployment acknowledged. Waiting to get processed: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,663] INFO marathon: DeploymentPlan 2015-09-03T15:44:01.647Z NO STEPS (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,664] INFO marathon:  (mesosphere.marathon.state.GroupManager:169) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,664] INFO marathon: [INFO] [09/03/2015 15:44:01.663]arathon-akka.actor.default-dispatcher-12] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,664] INFO marathon: [2015-09-03 15:44:01,663] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "DELETE //localhost:8140/v2/groups/marathonintegrationtest?force=true HTTP/1.1" 200 92 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$eeac8cb4:14) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,665] INFO marathon: [INFO] [09/03/2015 15:44:01.665] [pool-2-thread-2] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,665] INFO marathon: [INFO] [09/03/2015 15:44:01.665]arathon-akka.actor.default-dispatcher-9] [akka://marathon/user/MarathonScheduler/$a] Deployment of / successful (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,665] INFO marathon: [INFO] [09/03/2015 15:44:01.665]arathon-akka.actor.default-dispatcher-9] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,666] INFO marathon: [INFO] [09/03/2015 15:44:01.665]arathon-akka.actor.default-dispatcher-10] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager] Removing 3249f778-9761-4d65-9113-fa622a84cb58 from list of running deployments (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,667] INFO Received callback event: group_change_success with props Map(groupId -> /, version -> 2015-09-03T15:44:01.647Z, eventType -> group_change_success, timestamp -> 2015-09-03T15:44:01.663Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,668] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,669] INFO marathon: [INFO] [09/03/2015 15:44:01.669] [pool-2-thread-4] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,671] INFO Received callback event: deployment_success with props Map(id -> 3249f778-9761-4d65-9113-fa622a84cb58, plan -> Map(version -> 2015-09-03T15:44:01.647Z, id -> 3249f778-9761-4d65-9113-fa622a84cb58, target -> Map(groups -> List(), version -> 2015-09-03T15:44:01.647Z, id -> /, dependencies -> List(), apps -> List()), original -> Map(groups -> List(Map(groups -> List(), version -> 2015-09-03T15:44:01.384Z, id -> /marathonintegrationtest, dependencies -> List(), apps -> List())), version -> 2015-09-03T15:44:01.384Z, id -> /, dependencies -> List(), apps -> List()), steps -> List()), eventType -> deployment_success, timestamp -> 2015-09-03T15:44:01.665Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,672] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,771] INFO marathon: [2015-09-03 15:44:01,771] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "GET //localhost:8140/v2/apps HTTP/1.1" 200 11 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$eeac8cb4:14) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,774] INFO marathon: [2015-09-03 15:44:01,774] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "GET //localhost:8140/v2/groups HTTP/1.1" 200 87 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$eeac8cb4:14) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,783] INFO marathon: [2015-09-03 15:44:01,783] INFO validating ports array: List(0) (mesosphere.marathon.api.validation.PortsArrayValidator:21) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,785] INFO marathon: [2015-09-03 15:44:01,785] INFO Upgrade group id:/marathonintegrationtest version:1970-01-01T00:00:00.000Z with force:false (mesosphere.marathon.state.GroupManager:138) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,786] INFO marathon: [2015-09-03 15:44:01,786] INFO Take next configured free port: 10000 (mesosphere.marathon.state.GroupManager:213) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,787] INFO marathon: [2015-09-03 15:44:01,786] INFO [/marathonintegrationtest/app]: new app detected (mesosphere.marathon.upgrade.GroupVersioningUtil$:26) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,788] INFO marathon: [2015-09-03 15:44:01,788] INFO validating ports array: List(10000) (mesosphere.marathon.api.validation.PortsArrayValidator:21) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon: [2015-09-03 15:44:01,790] INFO Computed new deployment plan: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon: DeploymentPlan 1970-01-01T00:00:00.000Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon: step 1: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon:   * Start(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=0) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon: step 2: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,790] INFO marathon:   * Scale(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=1) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon:  (mesosphere.marathon.state.GroupManager:161) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon: [2015-09-03 15:44:01,790] INFO Deploy plan with force=false: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon: DeploymentPlan 1970-01-01T00:00:00.000Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon: step 1: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon:   * Start(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=0) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon: step 2: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,791] INFO marathon:   * Scale(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=1) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,792] INFO marathon:   (mesosphere.marathon.MarathonSchedulerService$$EnhancerByGuice$$7a577b4d:106) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,797] INFO marathon: [2015-09-03 15:44:01,793] INFO [/marathonintegrationtest/app] storing new app version 1970-01-01T00:00:00.000Z (mesosphere.marathon.state.GroupManager:144) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,803] INFO marathon: [INFO] [09/03/2015 15:44:01.803]arathon-akka.actor.default-dispatcher-5] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,805] INFO marathon: [2015-09-03 15:44:01,804] INFO Adding health check for app [/marathonintegrationtest/app] and version [1970-01-01T00:00:00.000Z]: [HealthCheck(Some(/),HTTP,0,None,20 seconds,1 second,20 seconds,10,false)] (mesosphere.marathon.health.MarathonHealthCheckManager$$EnhancerByGuice$$403214ad:61) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,805] INFO marathon: [INFO] [09/03/2015 15:44:01.804]arathon-akka.actor.default-dispatcher-13] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,805] INFO marathon: [INFO] [09/03/2015 15:44:01.805]arathon-akka.actor.default-dispatcher-11] [akka://marathon/user/$R] Starting health check actor for app [/marathonintegrationtest/app] and healthCheck [HealthCheck(Some(/),HTTP,0,None,20 seconds,1 second,20 seconds,10,false)] (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,807] INFO marathon: [2015-09-03 15:44:01,806] INFO Starting app /marathonintegrationtest/app (mesosphere.marathon.SchedulerActions:402) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,807] INFO marathon: [INFO] [09/03/2015 15:44:01.805] [pool-2-thread-1] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,807] INFO marathon: [INFO] [09/03/2015 15:44:01.806] [pool-2-thread-4] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,809] INFO Received callback event: add_health_check_event with props Map(healthCheck -> Map(timeoutSeconds -> 20, path -> /, portIndex -> 0, gracePeriodSeconds -> 20, maxConsecutiveFailures -> 10, ignoreHttp1xx -> false, intervalSeconds -> 1, protocol -> HTTP), eventType -> add_health_check_event, timestamp -> 2015-09-03T15:44:01.804Z, version -> 1970-01-01T00:00:00.000Z, appId -> /marathonintegrationtest/app) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,810] INFO mesos: I0903 15:44:01.810333  2663 master.cpp:4290] Sending 1 offers to framework 20150903-154311-16777343-5050-2654-0000 (marathon) at scheduler-52e7175e-2417-48eb-856e-ac8cd1917d99@172.17.0.2:53985 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,812] INFO Received callback event: deployment_info with props Map(plan -> Map(version -> 1970-01-01T00:00:00.000Z, id -> 3ba02384-356e-4b5e-98eb-fa04ae714970, target -> Map(groups -> List(Map(groups -> List(), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest, dependencies -> List(), apps -> List(Map(storeUrls -> List(), requirePorts -> false, disk -> 0.0, acceptedResourceRoles -> null, instances -> 1, container -> null, ports -> List(10000), cmd -> /tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1, uris -> List(), healthChecks -> List(Map(timeoutSeconds -> 20, path -> /, portIndex -> 0, gracePeriodSeconds -> 20, maxConsecutiveFailures -> 10, ignoreHttp1xx -> false, intervalSeconds -> 1, protocol -> HTTP)), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest/app, backoffFactor -> 1.15, labels -> Map(), constraints -> List(), mem -> 128.0, cpus -> 0.5, upgradeStrategy -> Map(minimumHealthCapacity -> 1.0, maximumOverCapacity -> 1.0), args -> null, maxLaunchDelaySeconds -> 3600, executor -> //cmd, dependencies -> List(), user -> null, env -> Map(), versionInfo -> Map(lastScalingAt -> 1970-01-01T00:00:00.000Z, lastConfigChangeAt -> 1970-01-01T00:00:00.000Z), backoffSeconds -> 1)))), version -> 1970-01-01T00:00:00.000Z, id -> /, dependencies -> List(), apps -> List()), original -> Map(groups -> List(), version -> 2015-09-03T15:44:01.647Z, id -> /, dependencies -> List(), apps -> List()), steps -> List(Map(actions -> List(Map(type -> StartApplication, app -> /marathonintegrationtest/app))), Map(actions -> List(Map(type -> ScaleApplication, app -> /marathonintegrationtest/app))))), currentStep -> Map(actions -> List(Map(type -> StartApplication, app -> /marathonintegrationtest/app))), eventType -> deployment_info, timestamp -> 2015-09-03T15:44:01.803Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,813] INFO mesos: I0903 15:44:01.813046  2663 master.cpp:2934] Processing REVIVE call for framework 20150903-154311-16777343-5050-2654-0000 (marathon) at scheduler-52e7175e-2417-48eb-856e-ac8cd1917d99@172.17.0.2:53985 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,813] INFO mesos: I0903 15:44:01.813166  2660 hierarchical.hpp:831] Removed filters for framework 20150903-154311-16777343-5050-2654-0000 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,813] INFO mesos: I0903 15:44:01.813091  2663 master.cpp:2884] Processing DECLINE call for offers: [ 20150903-154311-16777343-5050-2654-O14 ] for framework 20150903-154311-16777343-5050-2654-0000 (marathon) at scheduler-52e7175e-2417-48eb-856e-ac8cd1917d99@172.17.0.2:53985 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,814] INFO mesos: I0903 15:44:01.814014  2663 hierarchical.hpp:761] Recovered cpus(*):4; mem(*):14014; disk(*):32541; ports(*):[31000-32000] (total: cpus(*):4; mem(*):14014; disk(*):32541; ports(*):[31000-32000], allocated: ) on slave 20150903-154311-16777343-5050-2654-S0 from framework 20150903-154311-16777343-5050-2654-0000 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,814] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,814] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,815] INFO marathon: [2015-09-03 15:44:01,808] INFO Already running 0 instances of /marathonintegrationtest/app. Not scaling. (mesosphere.marathon.SchedulerActions:538) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,815] INFO marathon: [INFO] [09/03/2015 15:44:01.808]arathon-akka.actor.default-dispatcher-13] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager/3ba02384-356e-4b5e-98eb-fa04ae714970/$a] Successfully started 0 instances of /marathonintegrationtest/app (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,816] INFO marathon: [INFO] [09/03/2015 15:44:01.809]arathon-akka.actor.default-dispatcher-7] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,816] INFO marathon: [INFO] [09/03/2015 15:44:01.809]arathon-akka.actor.default-dispatcher-7] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,816] INFO marathon: [INFO] [09/03/2015 15:44:01.810]arathon-akka.actor.default-dispatcher-11] [akka://marathon/user/launchQueue/1/8-marathonintegrationtest_app] Started appTaskLaunchActor for /marathonintegrationtest/app version 1970-01-01T00:00:00.000Z with initial count 1 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,816] INFO marathon: [INFO] [09/03/2015 15:44:01.811]arathon-akka.actor.default-dispatcher-12] [akka://marathon/user/offerMatcherManager/1] activating matcher ActorOfferMatcher(Actor[akka://marathon/user/launchQueue/1/8-marathonintegrationtest_app#262927251]). (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,817] INFO marathon: [INFO] [09/03/2015 15:44:01.811]arathon-akka.actor.default-dispatcher-3] [akka://marathon/user/reviveOffersWhenWanted/1] Received offers WANTED notification (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,817] INFO marathon: [INFO] [09/03/2015 15:44:01.812]arathon-akka.actor.default-dispatcher-3] [akka://marathon/user/reviveOffersWhenWanted/1] => revive offers NOW, canceling any scheduled revives (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,817] INFO marathon: [INFO] [09/03/2015 15:44:01.816] [pool-2-thread-1] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,819] INFO Received callback event: deployment_step_success with props Map(plan -> Map(version -> 1970-01-01T00:00:00.000Z, id -> 3ba02384-356e-4b5e-98eb-fa04ae714970, target -> Map(groups -> List(Map(groups -> List(), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest, dependencies -> List(), apps -> List(Map(storeUrls -> List(), requirePorts -> false, disk -> 0.0, acceptedResourceRoles -> null, instances -> 1, container -> null, ports -> List(10000), cmd -> /tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1, uris -> List(), healthChecks -> List(Map(timeoutSeconds -> 20, path -> /, portIndex -> 0, gracePeriodSeconds -> 20, maxConsecutiveFailures -> 10, ignoreHttp1xx -> false, intervalSeconds -> 1, protocol -> HTTP)), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest/app, backoffFactor -> 1.15, labels -> Map(), constraints -> List(), mem -> 128.0, cpus -> 0.5, upgradeStrategy -> Map(minimumHealthCapacity -> 1.0, maximumOverCapacity -> 1.0), args -> null, maxLaunchDelaySeconds -> 3600, executor -> //cmd, dependencies -> List(), user -> null, env -> Map(), versionInfo -> Map(lastScalingAt -> 1970-01-01T00:00:00.000Z, lastConfigChangeAt -> 1970-01-01T00:00:00.000Z), backoffSeconds -> 1)))), version -> 1970-01-01T00:00:00.000Z, id -> /, dependencies -> List(), apps -> List()), original -> Map(groups -> List(), version -> 2015-09-03T15:44:01.647Z, id -> /, dependencies -> List(), apps -> List()), steps -> List(Map(actions -> List(Map(type -> StartApplication, app -> /marathonintegrationtest/app))), Map(actions -> List(Map(type -> ScaleApplication, app -> /marathonintegrationtest/app))))), currentStep -> Map(actions -> List(Map(type -> StartApplication, app -> /marathonintegrationtest/app))), eventType -> deployment_step_success, timestamp -> 2015-09-03T15:44:01.808Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,820] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,820] INFO marathon: [INFO] [09/03/2015 15:44:01.816] [pool-2-thread-4] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,821] INFO Received callback event: deployment_info with props Map(plan -> Map(version -> 1970-01-01T00:00:00.000Z, id -> 3ba02384-356e-4b5e-98eb-fa04ae714970, target -> Map(groups -> List(Map(groups -> List(), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest, dependencies -> List(), apps -> List(Map(storeUrls -> List(), requirePorts -> false, disk -> 0.0, acceptedResourceRoles -> null, instances -> 1, container -> null, ports -> List(10000), cmd -> /tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1, uris -> List(), healthChecks -> List(Map(timeoutSeconds -> 20, path -> /, portIndex -> 0, gracePeriodSeconds -> 20, maxConsecutiveFailures -> 10, ignoreHttp1xx -> false, intervalSeconds -> 1, protocol -> HTTP)), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest/app, backoffFactor -> 1.15, labels -> Map(), constraints -> List(), mem -> 128.0, cpus -> 0.5, upgradeStrategy -> Map(minimumHealthCapacity -> 1.0, maximumOverCapacity -> 1.0), args -> null, maxLaunchDelaySeconds -> 3600, executor -> //cmd, dependencies -> List(), user -> null, env -> Map(), versionInfo -> Map(lastScalingAt -> 1970-01-01T00:00:00.000Z, lastConfigChangeAt -> 1970-01-01T00:00:00.000Z), backoffSeconds -> 1)))), version -> 1970-01-01T00:00:00.000Z, id -> /, dependencies -> List(), apps -> List()), original -> Map(groups -> List(), version -> 2015-09-03T15:44:01.647Z, id -> /, dependencies -> List(), apps -> List()), steps -> List(Map(actions -> List(Map(type -> StartApplication, app -> /marathonintegrationtest/app))), Map(actions -> List(Map(type -> ScaleApplication, app -> /marathonintegrationtest/app))))), currentStep -> Map(actions -> List(Map(type -> ScaleApplication, app -> /marathonintegrationtest/app))), eventType -> deployment_info, timestamp -> 2015-09-03T15:44:01.809Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,821] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,834] INFO marathon: [2015-09-03 15:44:01,834] INFO Deployment acknowledged. Waiting to get processed: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,834] INFO marathon: DeploymentPlan 1970-01-01T00:00:00.000Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon: step 1: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon:   * Start(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=0) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:01][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon: step 2: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon:   * Scale(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";)), instances=1) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon:  (mesosphere.marathon.state.GroupManager:169) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,835] INFO marathon: [INFO] [09/03/2015 15:44:01.834]arathon-akka.actor.default-dispatcher-5] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,836] INFO marathon: [INFO] [09/03/2015 15:44:01.835]arathon-akka.actor.default-dispatcher-7] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,837] INFO marathon: [INFO] [09/03/2015 15:44:01.836] [pool-2-thread-6] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,838] INFO Received callback event: group_change_success with props Map(groupId -> /marathonintegrationtest, version -> 1970-01-01T00:00:00.000Z, eventType -> group_change_success, timestamp -> 2015-09-03T15:44:01.834Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,839] INFO marathon: [2015-09-03 15:44:01,839] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:8140/v2/apps HTTP/1.1" 201 923 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$eeac8cb4:14) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,841] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,842] INFO marathon: [INFO] [09/03/2015 15:44:01.841] [pool-2-thread-1] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,844] INFO Received callback event: api_post_event with props Map(clientIp -> 127.0.0.1, eventType -> api_post_event, timestamp -> 2015-09-03T15:44:01.834Z, uri -> /v2/apps, appDefinition -> Map(storeUrls -> List(), requirePorts -> false, disk -> 0.0, acceptedResourceRoles -> null, instances -> 1, container -> null, ports -> List(0), cmd -> /tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1, uris -> List(), healthChecks -> List(Map(timeoutSeconds -> 20, path -> /, portIndex -> 0, gracePeriodSeconds -> 20, maxConsecutiveFailures -> 10, ignoreHttp1xx -> false, intervalSeconds -> 1, protocol -> HTTP)), version -> 1970-01-01T00:00:00.000Z, id -> /marathonintegrationtest/app, backoffFactor -> 1.15, labels -> Map(), constraints -> List(), mem -> 128.0, cpus -> 0.5, upgradeStrategy -> Map(minimumHealthCapacity -> 1.0, maximumOverCapacity -> 1.0), args -> null, maxLaunchDelaySeconds -> 3600, executor -> //cmd, dependencies -> List(), user -> null, env -> Map(), backoffSeconds -> 1)) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:02][Step 3/3] [2015-09-03 15:44:01,844] INFO 127.0.0.1 - - [03/Sep/2015:15:44:01 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,535] INFO mesos: I0903 15:44:02.535578  2664 containerizer.cpp:1188] Executor for container '5074667b-0303-4c13-8122-7d8f571da0be' has exited (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,535] INFO mesos: I0903 15:44:02.535614  2664 containerizer.cpp:1001] Destroying container '5074667b-0303-4c13-8122-7d8f571da0be' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537140  2662 slave.cpp:3356] Executor 'marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002' of framework 20150903-154311-16777343-5050-2654-0000 exited with status 0 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537168  2662 slave.cpp:3460] Cleaning up executor 'marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002' of framework 20150903-154311-16777343-5050-2654-0000 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537442  2662 slave.cpp:3549] Cleaning up framework 20150903-154311-16777343-5050-2654-0000 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537534  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000/executors/marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002/runs/5074667b-0303-4c13-8122-7d8f571da0be' for gc 6.9999937808563days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537577  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000/executors/marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002' for gc 6.99999378032296days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537602  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/meta/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000/executors/marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002/runs/5074667b-0303-4c13-8122-7d8f571da0be' for gc 6.99999377992days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537624  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/meta/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000/executors/marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002' for gc 6.99999377968593days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537643  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000' for gc 6.99999377907556days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537662  2662 gc.cpp:56] Scheduling '/tmp/marathon-itest-mesos/0/meta/slaves/20150903-154311-16777343-5050-2654-S0/frameworks/20150903-154311-16777343-5050-2654-0000' for gc 6.99999377876148days in the future (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:02][Step 3/3] [2015-09-03 15:44:02,537] INFO mesos: I0903 15:44:02.537683  2662 status_update_manager.cpp:284] Closing status update streams for framework 20150903-154311-16777343-5050-2654-0000 (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:11][Step 3/3] 2015-09-03 15:44:11,317:1786(0x7f517db00700):ZOO_ERROR@handle_socket_error_msg@1697: Socket [127.0.0.1:2185] zk retcode=-4, errno=111(Connection refused): server refused to accept the client
[17:44:11][Step 3/3] [2015-09-03 15:44:11,769] INFO mesos: I0903 15:44:11.769131  2661 slave.cpp:3842] Current disk usage 2.83%. Max allowed age: 6.101888244533739days (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:11][Step 3/3] [2015-09-03 15:44:11,775] INFO mesos: I0903 15:44:11.775454  2664 slave.cpp:4179] Querying resource estimator for oversubscribable resources (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:11][Step 3/3] [2015-09-03 15:44:11,775] INFO mesos: I0903 15:44:11.775531  2664 slave.cpp:4193] Received oversubscribable resources  from the resource estimator (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:19][Step 3/3] [2015-09-03 15:44:19,331] INFO mesos: I0903 15:44:19.331564  2665 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_app.7ff77e61-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:20][Step 3/3] [2015-09-03 15:44:20,150] INFO mesos: I0903 15:44:20.149955  2664 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_app.807a42f2-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:21][Step 3/3] 2015-09-03 15:44:21,325:1786(0x7f517db00700):ZOO_ERROR@handle_socket_error_msg@1697: Socket [127.0.0.1:2185] zk retcode=-4, errno=111(Connection refused): server refused to accept the client
[17:44:25][Step 3/3] [2015-09-03 15:44:25,368] INFO mesos: I0903 15:44:25.368212  2660 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_app.8398c103-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:26][Step 3/3] [2015-09-03 15:44:25,904] INFO mesos: I0903 15:44:25.904546  2664 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_http-app.83e7cb64-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:26][Step 3/3] [2015-09-03 15:44:26,776] INFO mesos: I0903 15:44:26.776226  2662 slave.cpp:4179] Querying resource estimator for oversubscribable resources (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:26][Step 3/3] [2015-09-03 15:44:26,776] INFO mesos: I0903 15:44:26.776329  2661 slave.cpp:4193] Received oversubscribable resources  from the resource estimator (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:29][Step 3/3] [2015-09-03 15:44:29,094] INFO mesos: I0903 15:44:29.094576  2665 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_tcp-app.85d19a05-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:30][Step 3/3] [2015-09-03 15:44:30,348] INFO mesos: I0903 15:44:30.348772  2666 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_command-app.8692a016-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:31][Step 3/3] [2015-09-03 15:44:30,921] INFO mesos: I0903 15:44:30.920963  2660 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_app.86e294d8-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:31][Step 3/3] [2015-09-03 15:44:30,927] INFO mesos: I0903 15:44:30.927217  2664 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_app.86e0c017-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:31][Step 3/3] 2015-09-03 15:44:31,332:1786(0x7f517db00700):ZOO_ERROR@handle_socket_error_msg@1697: Socket [127.0.0.1:2185] zk retcode=-4, errno=111(Connection refused): server refused to accept the client
[17:44:31][Step 3/3] [2015-09-03 15:44:31,355] INFO mesos: I0903 15:44:31.355222  2664 slave.cpp:3757] Framework 20150903-154311-16777343-5050-2654-0000 seems to have exited. Ignoring registration timeout for executor 'marathonintegrationtest_failing.872b5da9-5252-11e5-afaa-0242ac110002' (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [INFO] [09/03/2015 15:44:31.840] [default-akka.actor.default-dispatcher-7] [akka://default/system/IO-TCP/selectors/$a/1] Message [akka.io.Tcp$Close$] from Actor[akka://default/user/IO-HTTP/group-0/1#1437218963] to Actor[akka://default/system/IO-TCP/selectors/$a/1#-933903725] was not delivered. [2] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[17:44:32][Step 3/3] [info] - update an app *** FAILED ***
[17:44:32][Step 3/3] [info]   java.lang.AssertionError: Waiting for event deployment_success to arrive took longer than 30 seconds. Give up.
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.setup.WaitTestSupport$.next$1(WaitTestSupport.scala:30)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.setup.WaitTestSupport$.waitFor(WaitTestSupport.scala:36)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEventMatching(MarathonCallbackTestSupport.scala:46)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEventMatching(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEventWith(MarathonCallbackTestSupport.scala:50)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEventWith(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEvent(MarathonCallbackTestSupport.scala:30)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEvent(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.AppDeployIntegrationTest$$anonfun$10.apply$mcV$sp(AppDeployIntegrationTest.scala:168)
[17:44:32][Step 3/3] [info]   at mesosphere.marathon.integration.AppDeployIntegrationTest$$anonfun$10.apply(AppDeployIntegrationTest.scala:163)
[17:44:32][Step 3/3] [info]   ...
[17:44:32][Step 3/3] [info]   + Given a new app 
[17:44:32][Step 3/3] [2015-09-03 15:44:31,908] INFO marathon: [2015-09-03 15:44:31,901] INFO Upgrade group id:/ version:2015-09-03T15:44:31.901Z with force:true (mesosphere.marathon.state.GroupManager:138) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: [2015-09-03 15:44:31,910] INFO Computed new deployment plan: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: DeploymentPlan 2015-09-03T15:44:31.901Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: step 1: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon:   * Stop(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";))) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon:  (mesosphere.marathon.state.GroupManager:161) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: [2015-09-03 15:44:31,911] INFO Deploy plan with force=true: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: DeploymentPlan 2015-09-03T15:44:31.901Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon: step 1: (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon:   * Stop(App(/marathonintegrationtest/app, cmd="/tmp/appProxy7239983180790583764.sh /marathonintegrationtest/app v1 http://localhost:11271/health/marathonintegrationtest/app/v1";))) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,911] INFO marathon:   (mesosphere.marathon.MarathonSchedulerService$$EnhancerByGuice$$7a577b4d:106) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: [ERROR] [09/03/2015 15:44:31.912]arathon-akka.actor.default-dispatcher-7] [akka://marathon/user/MarathonScheduler/$a] Deployment of / failed (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: mesosphere.marathon.DeploymentCanceledException: The upgrade has been cancelled (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at mesosphere.marathon.upgrade.DeploymentManager$$anonfun$receive$1$$anonfun$3.apply(DeploymentManager.scala:49) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at mesosphere.marathon.upgrade.DeploymentManager$$anonfun$receive$1$$anonfun$3.apply(DeploymentManager.scala:48) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at scala.collection.immutable.List.foreach(List.scala:381) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,914] INFO marathon: 	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at scala.collection.immutable.List.map(List.scala:285) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at mesosphere.marathon.upgrade.DeploymentManager$$anonfun$receive$1.applyOrElse(DeploymentManager.scala:48) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.actor.Actor$class.aroundReceive(Actor.scala:465) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at mesosphere.marathon.upgrade.DeploymentManager.aroundReceive(DeploymentManager.scala:21) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.actor.ActorCell.invoke(ActorCell.scala:487) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:254) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.dispatch.Mailbox.run(Mailbox.scala:221) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at akka.dispatch.Mailbox.exec(Mailbox.scala:231) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon: 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,915] INFO marathon:  (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,916] INFO marathon: [INFO] [09/03/2015 15:44:31.912]arathon-akka.actor.default-dispatcher-12] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,916] INFO marathon: [INFO] [09/03/2015 15:44:31.912]arathon-akka.actor.default-dispatcher-10] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager] Removing 3ba02384-356e-4b5e-98eb-fa04ae714970 from list of running deployments (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,916] INFO marathon: [INFO] [09/03/2015 15:44:31.912]arathon-akka.actor.default-dispatcher-13] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager] Conflicting deployments for deployment ae808a4f-e09d-4e81-b591-64996c2c73f0 have been canceled (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,916] INFO marathon: [INFO] [09/03/2015 15:44:31.913]arathon-akka.actor.default-dispatcher-4] [akka://marathon/user/launchQueue/1/8-marathonintegrationtest_app] Deregister as matcher. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,916] INFO marathon: [INFO] [09/03/2015 15:44:31.913]arathon-akka.actor.default-dispatcher-4] [akka://marathon/user/launchQueue/1/8-marathonintegrationtest_app] Stopped appTaskLaunchActor for /marathonintegrationtest/app version 1970-01-01T00:00:00.000Z (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,922] INFO marathon: [INFO] [09/03/2015 15:44:31.913]arathon-akka.actor.default-dispatcher-2] [akka://marathon/user/offerMatcherManager/1] removing matcher ActorOfferMatcher(Actor[akka://marathon/user/launchQueue/1/8-marathonintegrationtest_app#262927251]) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,922] INFO marathon: [INFO] [09/03/2015 15:44:31.913]arathon-akka.actor.default-dispatcher-2] [akka://marathon/user/reviveOffersWhenWanted/1] Received offers NOT WANTED notification, canceling any TimedChecks (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,922] INFO marathon: [INFO] [09/03/2015 15:44:31.914]arathon-akka.actor.default-dispatcher-4] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,922] INFO marathon: [INFO] [09/03/2015 15:44:31.914] [pool-2-thread-6] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,922] INFO marathon: [2015-09-03 15:44:31,916] INFO [/marathonintegrationtest/app] expunging app (mesosphere.marathon.state.GroupManager:147) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,926] INFO marathon: [INFO] [09/03/2015 15:44:31.922] [pool-2-thread-6] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,937] INFO marathon: [INFO] [09/03/2015 15:44:31.937]arathon-akka.actor.default-dispatcher-8] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,938] INFO marathon: [INFO] [09/03/2015 15:44:31.937]arathon-akka.actor.default-dispatcher-8] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager/ae808a4f-e09d-4e81-b591-64996c2c73f0/$a] Successfully killed all the tasks (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,939] INFO marathon: [2015-09-03 15:44:31,938] INFO Removing health check for app [/marathonintegrationtest/app] and version [1970-01-01T00:00:00.000Z]: [HealthCheck(Some(/),HTTP,0,None,20 seconds,1 second,20 seconds,10,false)] (mesosphere.marathon.health.MarathonHealthCheckManager$$EnhancerByGuice$$403214ad:85) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,939] INFO marathon: [2015-09-03 15:44:31,938] INFO Stopping app /marathonintegrationtest/app (mesosphere.marathon.SchedulerActions:409) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,940] INFO marathon: [INFO] [09/03/2015 15:44:31.939]arathon-akka.actor.default-dispatcher-12] [akka://marathon/user/$b] POSTing to all endpoints. (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,940] INFO marathon: [INFO] [09/03/2015 15:44:31.939]arathon-akka.actor.default-dispatcher-13] [akka://marathon/user/$R] Stopped health check actor for app [/marathonintegrationtest/app] and healthCheck [HealthCheck(Some(/),HTTP,0,None,20 seconds,1 second,20 seconds,10,false)] (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,941] INFO marathon: [2015-09-03 15:44:31,940] WARN App /marathonintegrationtest/app removed from TaskTracker (mesosphere.marathon.tasks.TaskTracker$$EnhancerByGuice$$f6c75f17:134) (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,941] INFO marathon: [INFO] [09/03/2015 15:44:31.941] [pool-2-thread-6] [akka://marathon/user/$b] Sending POST to:http://localhost:11271/callback (mesosphere.marathon.integration.setup.ProcessKeeper$:101)
[17:44:32][Step 3/3] - update an app *** FAILED *** (30 seconds, 102 milliseconds)
[17:44:32][Step 3/3]   java.lang.AssertionError: Waiting for event deployment_success to arrive took longer than 30 seconds. Give up.
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.setup.WaitTestSupport$.next$1(WaitTestSupport.scala:30)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.setup.WaitTestSupport$.waitFor(WaitTestSupport.scala:36)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEventMatching(MarathonCallbackTestSupport.scala:46)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEventMatching(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEventWith(MarathonCallbackTestSupport.scala:50)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEventWith(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.setup.MarathonCallbackTestSupport$class.waitForEvent(MarathonCallbackTestSupport.scala:30)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.waitForEvent(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest$$anonfun$10.apply$mcV$sp(AppDeployIntegrationTest.scala:168)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest$$anonfun$10.apply(AppDeployIntegrationTest.scala:163)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest$$anonfun$10.apply(AppDeployIntegrationTest.scala:163)
[17:44:32][Step 3/3]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[17:44:32][Step 3/3]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[17:44:32][Step 3/3]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[17:44:32][Step 3/3]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[17:44:32][Step 3/3]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:158)
[17:44:32][Step 3/3]   at org.scalatest.Suite$class.withFixture(Suite.scala:1121)
[17:44:32][Step 3/3]   at org.scalatest.FunSuite.withFixture(FunSuite.scala:1559)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$class.invokeWithFixture$1(FunSuiteLike.scala:155)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:167)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:167)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$class.runTest(FunSuiteLike.scala:167)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.org$scalatest$BeforeAndAfter$$super$runTest(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at org.scalatest.BeforeAndAfter$class.runTest(BeforeAndAfter.scala:200)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.runTest(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$runTests$1.apply(FunSuiteLike.scala:200)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$runTests$1.apply(FunSuiteLike.scala:200)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[17:44:32][Step 3/3]   at scala.collection.immutable.List.foreach(List.scala:381)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$class.runTests(FunSuiteLike.scala:200)
[17:44:32][Step 3/3]   at org.scalatest.FunSuite.runTests(FunSuite.scala:1559)
[17:44:32][Step 3/3]   at org.scalatest.Suite$class.run(Suite.scala:1423)
[17:44:32][Step 3/3]   at org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1559)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$run$1.apply(FunSuiteLike.scala:204)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$$anonfun$run$1.apply(FunSuiteLike.scala:204)
[17:44:32][Step 3/3]   at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
[17:44:32][Step 3/3]   at org.scalatest.FunSuiteLike$class.run(FunSuiteLike.scala:204)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.org$scalatest$BeforeAndAfterAllConfigMap$$super$run(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at org.scalatest.BeforeAndAfterAllConfigMap$class.liftedTree1$1(BeforeAndAfterAllConfigMap.scala:248)
[17:44:32][Step 3/3]   at org.scalatest.BeforeAndAfterAllConfigMap$class.run(BeforeAndAfterAllConfigMap.scala:247)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.org$scalatest$BeforeAndAfter$$super$run(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at org.scalatest.BeforeAndAfter$class.run(BeforeAndAfter.scala:241)
[17:44:32][Step 3/3]   at mesosphere.marathon.integration.AppDeployIntegrationTest.run(AppDeployIntegrationTest.scala:19)
[17:44:32][Step 3/3]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:444)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,946] INFO Received callback event: remove_health_check_event with props Map(appId -> /marathonintegrationtest/app, eventType -> remove_health_check_event, timestamp -> 2015-09-03T15:44:31.938Z) (mesosphere.marathon.integration.setup.CallbackEventHandler:88)
[17:44:32][Step 3/3]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:651)
[17:44:32][Step 3/3]   at sbt.ForkMain$Run$2.call(ForkMain.java:294)
[17:44:32][Step 3/3]   at sbt.ForkMain$Run$2.call(ForkMain.java:284)
[17:44:32][Step 3/3] [2015-09-03 15:44:31,946] INFO 127.0.0.1 - - [03/Sep/2015:15:44:31 +0000] "POST //localhost:11271/callback HTTP/1.1" 204 0 "-" "spray-can/1.3.2 (mesosphere.chaos.http.ChaosRequestLog:14)
[17:44:32][Step 3/3]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[17:44:32][Step 3/3]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[17:44:32][Step 3/3]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[17:44:32][Step 3/3]   at java.lang.Thread.run(Thread.java:745)
[17:44:32][Step 3/3]   + Given a new app 
```

--- a/changelog.md
+++ b/changelog.md
@@ -110,6 +110,14 @@ with the `reviveOffers` Mesos scheduler API call. This should result in more spe
 for any reason you dislike this behavior, you can disable it with `--disable_revive_offers_for_new_apps`
 (not recommended).
 
+The order in which mesos receives `reviveOffers` and `declineOffer` calls is not guaranteed. Therefore, as
+long as we still need offers to launch tasks, we repeat the `reviveOffers` call for `--revive_offers_repetitions`
+times so that our last `reviveOffers` will be received after all relevant `declineOffer` calls with high
+probability. 
+
+* `--revive_offers_repetitions` (Optional. Default: 3): 
+    Repeat every reviveOffer request this many times, delayed by the `--min_revive_offers_interval`.
+
 When Marathon has no current use for an offer, it will decline the offer for a configurable period. A short duration 
 might lead to resource starvation for other frameworks if you run many frameworks
 in your cluster. You should only need to reduce it if you use `--disable_revive_offers_for_new_apps`.

--- a/docs/docs/command-line-flags.md
+++ b/docs/docs/command-line-flags.md
@@ -168,6 +168,14 @@ invocations of this call.
 * <span class="label label-default">v0.11.0</span> `--min_revive_offers_interval` (Optional. Default: 5000): 
     Do not ask for all offers (also already seen ones) more often than this interval (ms).
     
+The order in which mesos receives `reviveOffers` and `declineOffer` calls is not guaranteed. Therefore, as
+long as we still need offers to launch tasks, we repeat the `reviveOffers` call for `--revive_offers_repetitions`
+times so that our last `reviveOffers` will be received after all relevant `declineOffer` calls with high
+probability. 
+
+* <span class="label label-default">v0.11.0</span> `--revive_offers_repetitions` (Optional. Default: 3): 
+    Repeat every reviveOffer request this many times, delayed by the `--min_revive_offers_interval`.
+
 If you want to disable calling reviveOffers (not recommended), you can use:
     
 * <span class="label label-default">v0.11.0</span> `--disable_revive_offers_for_new_apps`

--- a/src/main/scala/mesosphere/marathon/core/flow/ReviveOffersConfig.scala
+++ b/src/main/scala/mesosphere/marathon/core/flow/ReviveOffersConfig.scala
@@ -24,4 +24,8 @@ trait ReviveOffersConfig extends ScallopConf {
   lazy val minReviveOffersInterval = opt[Long]("min_revive_offers_interval",
     descr = "Do not ask for all offers (also already seen ones) more often than this interval (ms).",
     default = Some(5000))
+
+  lazy val reviveOffersRepetitions = opt[Int]("revive_offers_repetitions",
+    descr = "Repeat every reviveOffer request this many times, delayed by the --min_revive_offers_interval.",
+    default = Some(3))
 }

--- a/src/main/scala/mesosphere/marathon/core/flow/impl/ReviveOffersActor.scala
+++ b/src/main/scala/mesosphere/marathon/core/flow/impl/ReviveOffersActor.scala
@@ -9,6 +9,7 @@ import mesosphere.marathon.core.flow.impl.ReviveOffersActor.OffersWanted
 import mesosphere.marathon.event.{ SchedulerReregisteredEvent, SchedulerRegisteredEvent }
 import mesosphere.marathon.state.Timestamp
 import rx.lang.scala.{ Observable, Subscription }
+import scala.annotation.tailrec
 import scala.concurrent.duration._
 
 private[flow] object ReviveOffersActor {
@@ -26,16 +27,17 @@ private[flow] object ReviveOffersActor {
 /**
   * Revive offers whenever interest is signaled but maximally every 5 seconds.
   */
-private class ReviveOffersActor(
+private[impl] class ReviveOffersActor(
     clock: Clock, conf: ReviveOffersConfig,
     marathonEventStream: EventStream,
     offersWanted: Observable[Boolean],
     driverHolder: MarathonSchedulerDriverHolder) extends Actor with ActorLogging {
 
-  private[this] var subscription: Subscription = _
-  private[this] var offersCurrentlyWanted: Boolean = false
-  private[this] var lastRevive: Timestamp = Timestamp(0)
-  private[this] var nextReviveCancellableOpt: Option[Cancellable] = None
+  private[impl] var subscription: Subscription = _
+  private[impl] var offersCurrentlyWanted: Boolean = false
+  private[impl] var revivesNeeded: Int = 0
+  private[impl] var lastRevive: Timestamp = Timestamp(0)
+  private[impl] var nextReviveCancellableOpt: Option[Cancellable] = None
 
   override def preStart(): Unit = {
     subscription = offersWanted.map(OffersWanted(_)).subscribe(self ! _)
@@ -50,6 +52,7 @@ private class ReviveOffersActor(
     marathonEventStream.unsubscribe(self)
   }
 
+  @tailrec
   private[this] def reviveOffers(): Unit = {
     val now: Timestamp = clock.now()
     val nextRevive = lastRevive + conf.minReviveOffersInterval().milliseconds
@@ -61,6 +64,13 @@ private class ReviveOffersActor(
 
       driverHolder.driver.foreach(_.reviveOffers())
       lastRevive = now
+
+      revivesNeeded -= 1
+      if (revivesNeeded > 0) {
+        log.info("{} further revives still needed. Repeating reviveOffers according to --{} {}",
+          revivesNeeded, conf.reviveOffersRepetitions.name, conf.reviveOffersRepetitions())
+        reviveOffers()
+      }
     }
     else {
       lazy val untilNextRevive = now until nextRevive
@@ -86,26 +96,39 @@ private class ReviveOffersActor(
     case OffersWanted(true) =>
       log.info("Received offers WANTED notification")
       offersCurrentlyWanted = true
-      reviveOffers()
+      initiateNewSeriesOfRevives()
 
     case OffersWanted(false) =>
-      log.info("Received offers NOT WANTED notification, canceling any TimedChecks")
+      log.info("Received offers NOT WANTED notification, canceling {} revives", revivesNeeded)
       offersCurrentlyWanted = false
+      revivesNeeded = 0
       nextReviveCancellableOpt.foreach(_.cancel())
       nextReviveCancellableOpt = None
   }
 
+  def initiateNewSeriesOfRevives(): Unit = {
+    revivesNeeded = conf.reviveOffersRepetitions()
+    reviveOffers()
+  }
+
   private[this] def receiveReviveOffersEvents: Receive = {
-    case msg @ (_: SchedulerRegisteredEvent |
-      _: SchedulerReregisteredEvent |
-      ReviveOffersActor.TimedCheck |
-      OfferReviverDelegate.ReviveOffers) =>
+    case msg @ (_: SchedulerRegisteredEvent | _: SchedulerReregisteredEvent | OfferReviverDelegate.ReviveOffers) =>
+
       if (offersCurrentlyWanted) {
         log.info(s"Received reviveOffers notification: ${msg.getClass.getSimpleName}")
-        reviveOffers()
+        initiateNewSeriesOfRevives()
       }
       else {
         log.info(s"Ignoring ${msg.getClass.getSimpleName} because no one is currently interested in offers")
+      }
+
+    case ReviveOffersActor.TimedCheck =>
+      log.info(s"Received TimedCheck")
+      if (revivesNeeded > 0) {
+        reviveOffers()
+      }
+      else {
+        log.info("=> no revives needed right now")
       }
   }
 

--- a/src/test/scala/mesosphere/marathon/core/flow/impl/ReviveOffersActorTest.scala
+++ b/src/test/scala/mesosphere/marathon/core/flow/impl/ReviveOffersActorTest.scala
@@ -1,77 +1,117 @@
 package mesosphere.marathon.core.flow.impl
 
 import akka.actor._
-import akka.testkit.{ TestProbe, TestActorRef }
-import mesosphere.marathon.core.base.{ Clock, ConstantClock }
+import akka.testkit.{ TestActorRef, TestProbe }
+import mesosphere.marathon.core.base.ConstantClock
 import mesosphere.marathon.core.flow.ReviveOffersConfig
+import mesosphere.marathon.core.flow.impl.ReviveOffersActor.TimedCheck
 import mesosphere.marathon.event.{ SchedulerRegisteredEvent, SchedulerReregisteredEvent }
 import mesosphere.marathon.{ MarathonSchedulerDriverHolder, MarathonSpec }
 import org.apache.mesos.SchedulerDriver
-import org.mockito.{ Matchers, ArgumentMatcher, Mockito }
-import org.mockito.internal.matchers.{ Matches, CapturesArguments }
-import org.scalatest.GivenWhenThen
+import org.mockito.Mockito
+import org.scalatest.{ Matchers, GivenWhenThen }
 import rx.lang.scala.Subject
 import rx.lang.scala.subjects.PublishSubject
+
 import scala.concurrent.duration._
 
-class ReviveOffersActorTest extends MarathonSpec with GivenWhenThen {
+class ReviveOffersActorTest extends MarathonSpec with GivenWhenThen with Matchers {
   test("do not do anything") {
+    val f = new Fixture()
     When("the actor starts")
-    actorRef.start()
+    f.actorRef.start()
 
     Then("there are no surprising interactions")
-    verifyNoMoreInteractions()
+    f.verifyNoMoreInteractions()
   }
 
   test("revive on first OffersWanted(true)") {
+    val f = new Fixture()
     Given("a started actor")
-    actorRef.start()
+    f.actorRef.start()
 
     When("the actor gets notified of wanted offers")
-    offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
 
     Then("reviveOffers is called")
-    Mockito.verify(driver, Mockito.timeout(1000)).reviveOffers()
-    verifyNoMoreInteractions()
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+    f.verifyNoMoreInteractions()
   }
 
   test(s"revive if offers wanted and we receive explicit reviveOffers") {
+    val f = new Fixture()
     Given("a started actor that wants offers")
-    actorRef.start()
-    offersWanted.onNext(true)
-    Mockito.reset(driver)
-    clock += 10.seconds
+    f.actorRef.start()
+    f.offersWanted.onNext(true)
+    Mockito.reset(f.driver)
+    f.clock += 10.seconds
 
     When("we explicitly reviveOffers")
-    val offerReviver = new OfferReviverDelegate(actorRef)
+    val offerReviver = new OfferReviverDelegate(f.actorRef)
     offerReviver.reviveOffers()
 
     Then("reviveOffers is called directly")
-    Mockito.verify(driver, Mockito.timeout(1000)).reviveOffers()
-    verifyNoMoreInteractions()
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+    f.verifyNoMoreInteractions()
   }
 
   for (
     reviveEvent <- Seq(
       SchedulerReregisteredEvent("somemaster"),
-      SchedulerRegisteredEvent("frameworkid", "somemaster"),
-      ReviveOffersActor.TimedCheck
+      SchedulerRegisteredEvent("frameworkid", "somemaster")
     )
   ) {
     test(s"revive if offers wanted and we receive $reviveEvent") {
+      val f = new Fixture()
       Given("a started actor that wants offers")
-      actorRef.start()
-      offersWanted.onNext(true)
-      Mockito.reset(driver)
-      clock += 10.seconds
+      f.actorRef.start()
+      f.offersWanted.onNext(true)
+      Mockito.reset(f.driver)
+      f.clock += 10.seconds
 
       When(s"the actor receives an $reviveEvent message")
-      actorRef ! reviveEvent
+      f.actorRef ! reviveEvent
 
       Then("reviveOffers is called directly")
-      Mockito.verify(driver, Mockito.timeout(1000)).reviveOffers()
-      verifyNoMoreInteractions()
+      Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+      f.verifyNoMoreInteractions()
     }
+  }
+
+  test(s"NO revive if revivesWanted == 0 and we receive TimedCheck") {
+    val f = new Fixture()
+    Given("a started actor that wants offers")
+    f.actorRef.start()
+    f.offersWanted.onNext(true) // immediate revive, nothing scheduled
+    Mockito.reset(f.driver)
+    f.clock += 10.seconds
+
+    When(s"the actor receives an TimedCheck message")
+    f.actorRef ! TimedCheck
+
+    Then("reviveOffers is NOT called")
+    f.verifyNoMoreInteractions()
+  }
+
+  test(s"revive if revivesWanted > 0 and we receive TimedCheck") {
+    val f = new Fixture()
+    Given("a started actor that wants offers")
+    f.actorRef.start()
+    f.offersWanted.onNext(true) // immediate revive
+    f.offersWanted.onNext(true) // one revive scheduled
+    Mockito.reset(f.driver)
+    f.clock += 10.seconds
+
+    When(s"the actor receives an TimedCheck message")
+    f.actorRef ! TimedCheck
+
+    Then("reviveOffers is called directly")
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+
+    And("any scheduled timers are canceled")
+    Mockito.verify(f.actorRef.underlyingActor.cancellable, Mockito.timeout(1000)).cancel()
+
+    f.verifyNoMoreInteractions()
   }
 
   for (
@@ -82,132 +122,179 @@ class ReviveOffersActorTest extends MarathonSpec with GivenWhenThen {
     )
   ) {
     test(s"DO NOT revive if offers NOT wanted and we receive $reviveEvent") {
+      val f = new Fixture()
       Given("a started actor that wants offers")
-      actorRef.start()
-      offersWanted.onNext(false)
-      Mockito.reset(driver)
-      clock += 10.seconds
+      f.actorRef.start()
+      f.offersWanted.onNext(false)
+      Mockito.reset(f.driver)
+      f.clock += 10.seconds
 
       When(s"the actor receives an $reviveEvent message")
-      actorRef ! reviveEvent
+      f.actorRef ! reviveEvent
 
       Then("reviveOffers is NOT called directly")
-      verifyNoMoreInteractions()
+      f.verifyNoMoreInteractions()
     }
   }
 
   test("only one revive for two fast consecutive trues") {
+    val f = new Fixture()
     Given("a started actor")
-    actorRef.start()
+    f.actorRef.start()
 
     When("the actor gets notified twice at the same time of wanted offers")
-    offersWanted.onNext(true)
-    offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
 
     Then("it calls reviveOffers once")
-    Mockito.verify(driver, Mockito.timeout(1000)).reviveOffers()
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
     And("schedules the next revive for in 5 seconds")
-    assert(actorRef.underlyingActor.scheduled == Vector(5.seconds))
-    verifyNoMoreInteractions()
+    assert(f.actorRef.underlyingActor.scheduled == Vector(5.seconds))
+    f.verifyNoMoreInteractions()
   }
 
   test("the third true has no effect") {
+    val f = new Fixture()
     Given("a started actor")
-    actorRef.start()
+    f.actorRef.start()
 
     And("we already received two offers wanted notifications")
-    offersWanted.onNext(true)
-    offersWanted.onNext(true)
-    Mockito.reset(driver)
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
+    Mockito.reset(f.driver)
 
     When("we get another offers wanted 3 seconds later")
-    clock += 3.seconds
-    offersWanted.onNext(true)
+    f.clock += 3.seconds
+    f.offersWanted.onNext(true)
 
     Then("nothing happens because our next revive is already scheduled")
-    verifyNoMoreInteractions()
+    f.verifyNoMoreInteractions()
   }
 
   test("Revive timer is cancelled if offers not wanted anymore") {
+    val f = new Fixture()
     Given("we received offersWanted = true two times and thus scheduled a timer")
-    actorRef.start()
-    offersWanted.onNext(true)
-    offersWanted.onNext(true)
+    f.actorRef.start()
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
 
-    Mockito.reset(driver)
-    Mockito.reset(actorRef.underlyingActor.cancellable)
+    Mockito.reset(f.driver)
+    Mockito.reset(f.actorRef.underlyingActor.cancellable)
 
     When("we receive a false (= offers not wanted anymore) message")
-    offersWanted.onNext(false)
+    f.offersWanted.onNext(false)
 
     Then("we cancel the timer")
-    Mockito.verify(actorRef.underlyingActor.cancellable, Mockito.timeout(1000)).cancel()
-    verifyNoMoreInteractions()
+    Mockito.verify(f.actorRef.underlyingActor.cancellable, Mockito.timeout(1000)).cancel()
+    f.verifyNoMoreInteractions()
   }
 
   test("Check revives if last offersWanted == true and more than 5.seconds ago") {
+    val f = new Fixture()
     Given("that we received various flipping offers wanted requests")
-    actorRef.start()
-    offersWanted.onNext(true)
-    offersWanted.onNext(false)
-    offersWanted.onNext(true)
+    f.actorRef.start()
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(false)
+    f.offersWanted.onNext(true)
 
-    Mockito.reset(driver)
-    Mockito.reset(actorRef.underlyingActor.cancellable)
+    Mockito.reset(f.driver)
+    Mockito.reset(f.actorRef.underlyingActor.cancellable)
 
     And("we wait for 5 seconds")
-    clock += 5.seconds
+    f.clock += 5.seconds
 
     When("we receive a Check message")
-    actorRef ! ReviveOffersActor.TimedCheck
+    f.actorRef ! ReviveOffersActor.TimedCheck
 
     Then("we cancel our now unnecessary timer (which has send this message)")
-    Mockito.verify(actorRef.underlyingActor.cancellable, Mockito.timeout(1000)).cancel()
+    Mockito.verify(f.actorRef.underlyingActor.cancellable, Mockito.timeout(1000)).cancel()
     And("we revive the offers")
-    Mockito.verify(driver, Mockito.timeout(1000)).reviveOffers()
-    verifyNoMoreInteractions()
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+    f.verifyNoMoreInteractions()
   }
 
   test("Check does not revives if last offersWanted == false and more than 5.seconds ago") {
+    val f = new Fixture()
     Given("that we received various flipping offers wanted requests")
-    actorRef.start()
-    offersWanted.onNext(true)
-    offersWanted.onNext(true)
-    offersWanted.onNext(false)
+    f.actorRef.start()
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(true)
+    f.offersWanted.onNext(false)
 
-    Mockito.reset(driver)
-    Mockito.reset(actorRef.underlyingActor.cancellable)
+    Mockito.reset(f.driver)
+    Mockito.reset(f.actorRef.underlyingActor.cancellable)
 
     And("we wait for 5 seconds")
-    clock += 5.seconds
+    f.clock += 5.seconds
 
     When("we receive a Check message")
-    actorRef ! ReviveOffersActor.TimedCheck
+    f.actorRef ! ReviveOffersActor.TimedCheck
 
     Then("we do not do anything")
-    verifyNoMoreInteractions()
+    f.verifyNoMoreInteractions()
+  }
+
+  test("revive on repeatedly while OffersWanted(true)") {
+    val f = new Fixture(repetitions = 5)
+    Given("a started actor")
+    f.actorRef.start()
+
+    When("the actor gets notified of wanted offers")
+    f.offersWanted.onNext(true)
+
+    Then("reviveOffers is called and we schedule more revives")
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+    And("we scheduled the first of many revives")
+    f.actorRef.underlyingActor.scheduled should have size (1)
+    f.actorRef.underlyingActor.revivesNeeded should be (f.repetitions - 1)
+
+    Mockito.reset(f.driver)
+
+    for (i <- 2L to f.repetitions.toLong - 1) {
+      When("the min_revive_offers_interval has passed and we receive a TimedCheck")
+      f.clock += f.conf.minReviveOffersInterval().millis
+      f.actorRef ! ReviveOffersActor.TimedCheck
+
+      Then("reviveOffers is called")
+      Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+      Mockito.verifyNoMoreInteractions(f.driver)
+      Mockito.reset(f.driver)
+
+      And("current timer gets canceled")
+      Mockito.verify(f.actorRef.underlyingActor.cancellable).cancel()
+      Mockito.verifyNoMoreInteractions(f.actorRef.underlyingActor.cancellable)
+      Mockito.reset(f.actorRef.underlyingActor.cancellable)
+
+      And("we have scheduled the next revive")
+      f.actorRef.underlyingActor.scheduled should have size i
+      f.actorRef.underlyingActor.revivesNeeded should be (f.repetitions - i)
+    }
+
+    When("the min_revive_offers_interval has passed and we receive our last TimedCheck")
+    f.clock += f.conf.minReviveOffersInterval().millis
+    f.actorRef ! ReviveOffersActor.TimedCheck
+
+    Then("reviveOffers is called for the last time")
+    Mockito.verify(f.driver, Mockito.timeout(1000)).reviveOffers()
+    Mockito.verifyNoMoreInteractions(f.driver)
+    Mockito.reset(f.driver)
+
+    And("current timer gets canceled")
+    Mockito.verify(f.actorRef.underlyingActor.cancellable).cancel()
+    Mockito.verifyNoMoreInteractions(f.actorRef.underlyingActor.cancellable)
+    Mockito.reset(f.actorRef.underlyingActor.cancellable)
+
+    And("we have NOT scheduled the next revive")
+    f.actorRef.underlyingActor.scheduled should have size (f.repetitions.toLong - 1)
+    f.actorRef.underlyingActor.revivesNeeded should be (0)
+
+    f.verifyNoMoreInteractions()
   }
 
   private[this] implicit var actorSystem: ActorSystem = _
-  private[this] val conf = new ReviveOffersConfig {}
-  conf.afterInit()
-  private[this] var clock: ConstantClock = _
-  private[this] var actorRef: TestActorRef[TestableActor] = _
-  private[this] var offersWanted: Subject[Boolean] = _
-  private[this] var driver: SchedulerDriver = _
-  private[this] var driverHolder: MarathonSchedulerDriverHolder = _
-  private[this] var mockScheduler: Scheduler = _
 
   before {
     actorSystem = ActorSystem()
-    clock = ConstantClock()
-    offersWanted = PublishSubject()
-    driver = mock[SchedulerDriver]
-    driverHolder = new MarathonSchedulerDriverHolder
-    driverHolder.driver = Some(driver)
-    mockScheduler = mock[Scheduler]
-
-    actorRef = TestActorRef(new TestableActor)
   }
 
   after {
@@ -215,31 +302,54 @@ class ReviveOffersActorTest extends MarathonSpec with GivenWhenThen {
     actorSystem.awaitTermination()
   }
 
-  private[this] def verifyNoMoreInteractions(): Unit = {
-    def killActorAndWaitForDeath(): Terminated = {
-      actorRef ! PoisonPill
-      val deathWatch = TestProbe()
-      deathWatch.watch(actorRef)
-      deathWatch.expectMsgClass(classOf[Terminated])
+  class Fixture(val repetitions: Int = 1) {
+    lazy val conf: ReviveOffersConfig = {
+      val conf = new ReviveOffersConfig {
+        override lazy val reviveOffersRepetitions = opt[Int]("revive_offers_repetitions",
+          descr = "Repeat every reviveOffer request this many times, delayed by the --min_revive_offers_interval.",
+          default = Some(repetitions))
+      }
+      conf.afterInit()
+      conf
+    }
+    lazy val clock: ConstantClock = ConstantClock()
+    lazy val offersWanted: Subject[Boolean] = PublishSubject()
+    lazy val driver: SchedulerDriver = mock[SchedulerDriver]
+    lazy val driverHolder: MarathonSchedulerDriverHolder = {
+      val holder = new MarathonSchedulerDriverHolder
+      holder.driver = Some(driver)
+      holder
+    }
+    lazy val mockScheduler: Scheduler = mock[Scheduler]
+    lazy val actorRef = TestActorRef(new TestableActor)
+
+    def verifyNoMoreInteractions(): Unit = {
+      def killActorAndWaitForDeath(): Terminated = {
+        actorRef ! PoisonPill
+        val deathWatch = TestProbe()
+        deathWatch.watch(actorRef)
+        deathWatch.expectMsgClass(classOf[Terminated])
+      }
+
+      Mockito.verifyNoMoreInteractions(actorRef.underlyingActor.cancellable)
+
+      killActorAndWaitForDeath()
+
+      Mockito.verifyNoMoreInteractions(driver)
+      Mockito.verifyNoMoreInteractions(mockScheduler)
     }
 
-    Mockito.verifyNoMoreInteractions(actorRef.underlyingActor.cancellable)
+    class TestableActor extends ReviveOffersActor(
+      clock, conf, actorSystem.eventStream, offersWanted, driverHolder
+    ) {
+      var scheduled = Vector.empty[FiniteDuration]
+      var cancellable = mock[Cancellable]
 
-    killActorAndWaitForDeath()
-
-    Mockito.verifyNoMoreInteractions(driver)
-    Mockito.verifyNoMoreInteractions(mockScheduler)
-  }
-
-  private class TestableActor extends ReviveOffersActor(
-    clock, conf, actorSystem.eventStream, offersWanted, driverHolder
-  ) {
-    var scheduled = Vector.empty[FiniteDuration]
-    var cancellable = mock[Cancellable]
-
-    override protected def schedulerCheck(duration: FiniteDuration): Cancellable = {
-      scheduled :+= duration
-      cancellable
+      override protected def schedulerCheck(duration: FiniteDuration): Cancellable = {
+        scheduled :+= duration
+        cancellable
+      }
     }
+
   }
 }

--- a/src/test/scala/mesosphere/marathon/integration/GroupDeployIntegrationTest.scala
+++ b/src/test/scala/mesosphere/marathon/integration/GroupDeployIntegrationTest.scala
@@ -1,6 +1,7 @@
 package mesosphere.marathon.integration
 
 import mesosphere.marathon.api.v2.json.{ V2AppDefinition, V2GroupUpdate }
+import mesosphere.marathon.health.HealthCheck
 import mesosphere.marathon.integration.setup.{ IntegrationFunSuite, IntegrationHealthCheck, SingleMarathonIntegrationTest, WaitTestSupport }
 import mesosphere.marathon.state.{ PathId, UpgradeStrategy }
 import org.apache.http.HttpStatus

